menuconfig.py

```diff
@@ -299,6 +299,141 @@
+    """
+    # The formula for converting a single RGB value to the 6cube index
+    rto6 = lambda x: 0 if x < 48 else max(1, int(round((x - 55) / 40)))
+
+    return [rto6(x) for x in rgb]
+
+def rgb_from_6cube(c6):
+    """
+    Convert a 6cube index triplet to the corresponding RGB value
+    """
+    # The formula from converting the xterm 6cube index to a RGB value
+    rfrom6 = lambda x: 0 if x == 0 else 55 + x*40
+    return [rfrom6(x) for x in c6]
+
+def rgb_to_gridx(rgb):
+    """
+    Convert a RGB color to its closest xterm grayscale color
+    (indexes from 0-23 represents xterm colros 232-255)
+    """
+    gray = int(round((rgb[0] + rgb[1] + rgb[2]) / 3))
+
+    # Convert the single gray value component to the xterm greyscale index
+    # (relative to 232)
+    idx = int(round((gray - 8) / 10))
+
+    # Clamp the value between 0 and 23
+    return max(0, min(idx, 23))
+
+def rgb_from_gridx(c):
+    """
+    Convert a grayscale index to its closet single RGB component
+    """
+    g = c * 10 + 8
+    return (g, g, g)
+
+def rgb_alloc(rgb, RGB_COLORS={}):
+    """
+    Ncurses doesn't allow you to define new colors -- you are allowed only
+    to overwrite existing ones.
+
+    The colors from 0-15 are user-defined and there's no way to query the
+    RGB values so we better leave them untouched.
+
+    The RGB values of colors from 16-255 can be easily calculated. However,
+    colors from 232-255 use a different formula (grayscale) so stick with
+    colors from 16-231. 200+ colors should be plentiful.
+    """
+    if rgb in RGB_COLORS:
+        return RGB_COLORS[rgb]
+
+    # The first 16 colors are user defined and we should not change their
+    # values
+    cn = len(RGB_COLORS) + 16
+    if cn >= curses.COLORS:
+        raise StyleError("Unable to allocate new RGB color.")
+
+    # Formula from converting a RGB value to a curses color
+    # It's really weird that they chose 1000 as a max value
+    ctocr = lambda x: int(round(x * 1000 / 255))
+    curses.init_color(cn, *[ctocr(x) for x in rgb])
+    RGB_COLORS[rgb] = cn
+
+    return cn
+
+def color_get(num):
+    # Simplest case -- terminal doesn't support RGB colors
+    # color_get_rgb() won't be changing the current palettte
+    # so we can return the color as-is
+    if not curses.can_change_color():
+        return num
+
+    # Standard colors, rgb_alloc() doesn't touch these so
+    # we can return them as-is
+    if num < 16 or num >= 232:
+        return num
+
+    # We're in RGB mode. rgb_alloc() will redefine the colors
+    # from 16-231. Since the RGB value of the "standard" 256 color
+    # palette can be easily calculated, we can emulate them in RGB
+    # mode rather easily
+    num -= 16
+    rgb = rgb_from_6cube((num / 36) % 6, (num / 6) % 6, num % 6)
+
+    return rgb_alloc(rgb)
+
+def color_get_rgb(rgb):
+    # Lambda for calculating the distance between two RGB colors
+    dist = lambda r1, r2: sum([(x - y)**2 for x, y in zip(r1, r2)])
```

Comment:
Could have `sum((x - y)**2 for x, y in zip(r1, r2))`, using a generator instead of a list comprehension.

Maybe the comment could say `...for calculating the (squared) distance...`, just to be clear.

Reply:
Changed to use the term "Euclidean" distance.